src/components/song.tsx

```diff
@@ -0,0 +1,46 @@
+import {Avatar, Icon, List} from 'antd';
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {getSong} from '../reducers/songs';
+
+class SongView extends React.Component<{
+  id: number,
+  name?: string,
+  avatar?: string,
+  singer?: string,
+  isActive?: boolean,
+  playAction: (songID: number) => (event: any) => void
+}> {
+
+  public getIcons = () => {
```

Comment:
Usually when you try to extract a function inside a react comp, what you really want is just another componenet, so here that will be `<PlayPauseIcon onClick=...>`